pd/src/info/light_wallet.rs

```diff
@@ -0,0 +1,60 @@
+use penumbra_proto::{
+    chain::{ChainParams, CompactBlock},
+    light_wallet::{
+        light_wallet_server::LightWallet, ChainParamsRequest, CompactBlockRangeRequest,
+        ValidatorInfoRequest,
+    },
+    stake::ValidatorInfo,
+};
+use std::pin::Pin;
+use tonic::Status;
+use tracing::instrument;
+
+use crate::WriteOverlayExt;
+
+struct WalletOverlay<T: WriteOverlayExt>(T);
+
+#[tonic::async_trait]
+impl<T> LightWallet for WalletOverlay<T>
+where
+    T: 'static + WriteOverlayExt + Send + Sync,
+{
```

Comment:
I think we should be able to avoid having to have the wrapper struct by doing something like
```
impl<T: WriteOverlayExt> LightWallet for T
```
The `Send + Sync` bounds are already implied by `WriteOverlayExt`, and if we need it to be `'static` here, we could add a `'static` bound to `WriteOverlayExt` so that it's available for every user

Reply:
I tried this originally & it was in violation of the orphan rules 🤔

Reply:
ah boo

Reply:
i'm surprised because `WriteOverlayExt` is defined in-crate. but in that case, maybe it would be better to have one wrapper struct instead of two, and lift it into the parent `info` module as an `RpcWrapper`, that we can then add whatever extra extension traits to